samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/removingproductitem/RemoveProductItemFromShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package io.eventdriven.ecommerce.shoppingcarts.removingproductitem;
+
+import io.eventdriven.ecommerce.shoppingcarts.Events;
+import io.eventdriven.ecommerce.shoppingcarts.ShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.productitems.PricedProductItem;
+
+import java.util.UUID;
+
+public record RemoveProductItemFromShoppingCart(
+  UUID shoppingCartId,
```

Comment:
I'd recommend dedicated value object for ids like `ShoppingCartId` for do not mismatch with other UUIds etc.

Reply:
I'm not a huge fan of strongly typed ids; they add more ceremony than they're bringing value. In more advanced cases, this may help, but I feel that this may be an overkill for this example. I prefer to have that covered through tests (which, of course, I don't have right now in my sample, but I'm planning to add them 😅 ).